src/test/trackexport_test.cpp

```diff
@@ -0,0 +1,283 @@
+    EXPECT_EQ(2, m_answerer->currentProgressCount());
+
+    // Both files should have been overwritten.
+    QFileInfo newfile1(m_exportDir.filePath("cover-test.ogg"));
+    EXPECT_TRUE(newfile1.exists());
+    EXPECT_EQ(110106, newfile1.size());
+
+    QFileInfo newfile2(m_exportDir.filePath("cover-test.m4a"));
+    EXPECT_TRUE(newfile2.exists());
+    EXPECT_EQ(373820, newfile2.size());
+}
+
+TEST_F(TrackExporterTest, SkipAll) {
+    // Export a tracklist with two existing tracks -- skip both.
+    QFileInfo fileinfo1(m_testDataDir.filePath("cover-test.ogg"));
+    TrackPointer track1(new TrackInfoObject(fileinfo1));
+    QFileInfo fileinfo2(m_testDataDir.filePath("cover-test.m4a"));
+    TrackPointer track2(new TrackInfoObject(fileinfo2));
+
+    // Create empty versions at the destination so we can see if we actually
+    // overwrote or skipped.
+    QFile file1(m_exportDir.filePath("cover-test.ogg"));
+    ASSERT_TRUE(file1.open(QIODevice::WriteOnly));
+    file1.close();
+    QFile file2(m_exportDir.filePath("cover-test.m4a"));
+    ASSERT_TRUE(file2.open(QIODevice::WriteOnly));
+    file2.close();
+
+    // Set up the worker and answerer.
+    QList<TrackPointer> tracks{track1, track2};
+    TrackExportWorker worker(m_exportDir.canonicalPath(), tracks);
+    m_answerer.reset(new FakeOverwriteAnswerer(&worker));
+    // This could be flaky if the map key order is non-deterministic.
+    m_answerer->setAnswer(m_exportDir.filePath("cover-test.m4a"),
+                           TrackExportWorker::OverwriteAnswer::SKIP_ALL);
+
+    worker.run();
+    EXPECT_TRUE(worker.wait(10000));
+
+    EXPECT_EQ(2, m_answerer->currentProgress());
+    EXPECT_EQ(2, m_answerer->currentProgressCount());
+
+    // Both files should have been skipped.
+    QFileInfo newfile1(m_exportDir.filePath("cover-test.ogg"));
+    EXPECT_TRUE(newfile1.exists());
+    EXPECT_EQ(0, newfile1.size());
+
+    QFileInfo newfile2(m_exportDir.filePath("cover-test.m4a"));
+    EXPECT_TRUE(newfile2.exists());
+    EXPECT_EQ(0, newfile2.size());
+}
+
+TEST_F(TrackExporterTest, Cancel) {
+    // Export a tracklist with two existing tracks, but cancel before we do
+    // anything.
+    QFileInfo fileinfo1(m_testDataDir.filePath("cover-test.ogg"));
+    TrackPointer track1(new TrackInfoObject(fileinfo1));
+    QFileInfo fileinfo2(m_testDataDir.filePath("cover-test.m4a"));
+    TrackPointer track2(new TrackInfoObject(fileinfo2));
+
+    // Create empty version at the destination so we can see if we actually
+    // canceled.
+    QFile file2(m_exportDir.filePath("cover-test.m4a"));
+    ASSERT_TRUE(file2.open(QIODevice::WriteOnly));
+    file2.close();
+
+    // Set up the worker and answerer.
+    QList<TrackPointer> tracks{track1, track2};
+    TrackExportWorker worker(m_exportDir.canonicalPath(), tracks);
+    m_answerer.reset(new FakeOverwriteAnswerer(&worker));
+    // This could be flaky if the map key order is non-deterministic.
```

Comment:
I was thinking one could write a test where a filename was duplicated, but that won't happen because of the munging.  So instead it would be a bug to try to have two answers with the same filename in a test